### PR TITLE
Rename classes and move them to the right namespace

### PR DESCRIPTION
**Pull request by tvdijen**
No functional changes

### CODE REVIEW

File: src/SAML2/HTTPArtifact.php [hunk 1]
```diff
@@ -169,7 +172,7 @@ public function receive(): Message
             throw new \Exception('Empty ArtifactResponse received, maybe a replay?');
         }
 
-        $samlResponse = Message::fromXML($xml);
+        $samlResponse = AbstractMessage::fromXML($xml);
```
Comment (jaimeperez) [inline src/SAML2/HTTPArtifact.php]: This is the method we agreed to move to an standalone factory.

File: src/SAML2/XML/samlp/AbstractMessage.php [hunk 2]
```diff
@@ -431,8 +420,10 @@ public function setRelayState(string $relayState = null): void
      *
      * @return \DOMElement The root element of the DOM tree
      */
-    public function toXML(): DOMElement
+    public function toXML(?DOMElement $parent = null): DOMElement
     {
+        Assert::null($parent);
```
Comment (jaimeperez) [inline src/SAML2/XML/samlp/AbstractMessage.php]: I think there can be situations where a message is not the actual root of the XML document (SOAP? PAOS?), so we should actually use `$parent` here.

File: src/SAML2/XML/samlp/AbstractStatusResponse.php [hunk 3]
```diff
@@ -132,16 +132,18 @@ public function setStatus(Status $status): void
      *
      * @return \DOMElement This status response.
      */
-    public function toXML(): DOMElement
+    public function toXML(?DOMElement $parent = null): DOMElement
     {
-        $root = parent::toXML();
+        Assert::null($parent);
```
Comment (jaimeperez) [inline src/SAML2/XML/samlp/AbstractStatusResponse.php]: Same here.

File: src/SAML2/XML/samlp/AbstractSubjectQuery.php [hunk 4]
```diff
@@ -109,17 +112,18 @@ public function setNameId(NameID $nameId): void
      *
      * @throws \InvalidArgumentException if assertions are false
      */
-    public function toXML(): DOMElement
+    public function toXML(?DOMElement $parent = null): DOMElement
     {
+        Assert::null($parent);
```
Comment (jaimeperez) [inline src/SAML2/XML/samlp/AbstractSubjectQuery.php]: And here.

**Comment by tvdijen** [comment]
As discussed the comments will be addressed in the next PR where we migrate the classes to the new API

**Pull request merged**